### ISSUE TITLE
[Inference] Implement predictor function with confidence scores

### ISSUE THREAD

**Issue by luipin** [luipin/set-ml-module#5]
## Goal

Implement a `predict()` function (or class) that takes an image and returns a human-readable prediction with per-feature confidence scores.

## Requirements (from spec Milestone 4)

- **Input:** Path to an image file (or a PIL/numpy image)
- **Output:** JSON-serializable dict, e.g.:
  ```json
  {
    "color":   {"prediction": "red",      "confidence": 0.97},
    "shape":   {"prediction": "diamond",  "confidence": 0.88},
    "number":  {"prediction": "2",        "confidence": 0.95},
    "shading": {"prediction": "solid",    "confidence": 0.91}
  }
  ```
- Apply `softmax` to logits to get probabilities; report the max as confidence
- Reuse `get_val_transforms()` for preprocessing
- Handle the inverse label mapping (integer → human-readable string)
- Google-style docstrings

## Files
- `src/models/multi_head_resnet.py` (or a new `src/models/predictor.py`)

## Depends on
- Issue #2 (model architecture)
- Issue #1 (resize fix, so preprocessing matches training)
